modules/git/git.go

```diff
@@ -8,6 +8,10 @@ import (
 	"github.com/gruntwork-io/terratest/modules/testing"
 )
 
+// Separate variables created to be able to mock commands in tests
+var branchNameExecCommand = exec.Command
+var tagExecCommand = exec.Command
```

Comment:
I think we should run the actual command during testing. You can still test this accordingly by cloning a copy of terratest to a temporary folder and checking out various branches and tags.

Reply:
Sure, makes sense. Will give it a try with proposed approach

Reply:
@yorinasub17 Performance results are not so good for terratest temporary repo clone (depends on the connection but on my end git module test execution lasts for around 20s). Is that acceptable or should we change the approach?